src/context.ts

```diff
@@ -0,0 +1,59 @@
+import React, {
+  createContext as reactCreateContext,
+  useContext,
+  useRef,
+} from 'react'
+import create, { UseStore } from 'zustand'
+import { EqualityChecker, State, StateCreator, StateSelector } from './vanilla'
+
+function createContext<TState extends State>(initialState?: TState) {
+  const ZustandContext = reactCreateContext<UseStore<TState> | undefined>(
+    undefined
+  )
+
+  const Provider = ({
+    createState,
+    children,
+  }: {
+    createState: StateCreator<TState>
+    children: React.ReactNode
+  }) => {
+    const storeRef = useRef<UseStore<TState>>()
+
+    if (!storeRef.current) {
+      storeRef.current = create(createState)
+    }
+
+    return React.createElement(
```

Comment:
```suggestion
    return createElement(
```

Just a minor thing, but please import it directly.